src/app/item-page/simple/field-components/specific-field/metrics/altmetric/item-page-altmetric-field.component.ts

```diff
@@ -0,0 +1,62 @@
+import { AfterViewInit, Component, EventEmitter, HostListener, Inject, Input, Output } from '@angular/core';
+import { ExternalScriptLoaderService } from 'src/app/shared/utils/scripts-loader/external-script-loader.service';
+import {
+  ExternalScriptsNames,
+  ExternalScriptsStatus,
+} from 'src/app/shared/utils/scripts-loader/external-script.model';
+import { Item } from '../../../../../../core/shared/item.model';
+import { APP_CONFIG, AppConfig } from 'src/config/app-config.interface';
+
+@Component({
+  selector: 'ds-item-page-altmetric-field',
+  templateUrl: './item-page-altmetric-field.component.html',
+})
+export class ItemPageAltmetricFieldComponent implements AfterViewInit {
+  @Input() item: Item;
+
+  @Output() widgetLoaded = new EventEmitter<boolean>();
+
+  constructor(
+    @Inject(APP_CONFIG) protected appConfig: AppConfig,
+    private scriptLoader: ExternalScriptLoaderService
+  ) {}
+
+  ngAfterViewInit() {
+    if (!this.appConfig.item.showAltmetricBadge) {
+      return;
+    }
+
+    this.scriptLoader
+      .load(ExternalScriptsNames.ALTMETRIC)
+      .then((data) => this.reloadBadge(data))
+      .catch((error) => console.error(error));
+  }
+
+  /**
+   * We ensure that the badge is visible after the script is loaded
+   * @param data The data returned from the promise
+   */
+  private reloadBadge(data: any[]) {
```

Comment:
I just want to point out that I was trying to implement this, but instead of for Altmetrics, for Dimensions. I basically cp/pt'd the files that were Altmetric specific, but renamed them and some of the variables/constants in the code to "dimensions."  I had it working except that when I left the item page and then came back to the page the badge was gone.  The issue came from this method.  I thought that I could use _dimensison_embed_init as the method, but not that did not work.  That method does not exists in the Dimensions' javascript.  I reached out to the community and @sergius02 told me to change this method to this:

  private reloadBadge(data: any[]) {
    if (data.find((element) => this.isLoaded(element))) {
      const initClass = '__dimensions_embed';
      const initMethod = 'addBadges';
      window[initClass][initMethod]();
    }
  }

Then it worked.

Reply:
@sergius02, I've tried this method but for Plumx. The [PlumX](https://plu.mx/plum/developers/widgets#Adding-Widgets-Dynamically) developer page says to call this method `window.__plumX.widgets.init();` but unfortunately if I navigate away from the page and return later, I received a `TypeError: window[initClass] is undefined` error. I tried replacing the value of initClass above with `__plumX.widgets.init` but the error persists. Any ideas how to resolve this issue?